src/internal/plugins/helpers/use-discovered-action.tsx

```diff
@@ -25,36 +35,34 @@ function RuntimeActionWrapper({ mountContent, unmountContent, context }: Runtime
   return <div ref={ref}></div>;
 }
 
-function convertRuntimeAction(action: ActionConfig | null, context: ActionContext) {
-  if (!action) {
-    return null;
-  }
-  return (
-    <RuntimeActionWrapper
-      key={action.id + '-' + context.type}
-      context={context}
-      mountContent={action.mountContent}
-      unmountContent={action.unmountContent}
-    />
-  );
-}
-
 export function createUseDiscoveredAction(onActionRegistered: ActionButtonsController['onActionRegistered']) {
   return function useDiscoveredAction(type: string): {
     discoveredActions: React.ReactNode[];
     headerRef: React.Ref<HTMLDivElement>;
     contentRef: React.Ref<HTMLDivElement>;
   } {
-    const [discoveredActions, setDiscoveredActions] = useState<Array<React.ReactNode>>([]);
+    const [actionConfigs, setActionConfigs] = useState<Array<ActionConfig>>([]);
     const headerRef = useRef<HTMLDivElement>(null);
     const contentRef = useRef<HTMLDivElement>(null);
 
     useEffect(() => {
-      return onActionRegistered(actions => {
-        setDiscoveredActions(actions.map(action => convertRuntimeAction(action, { type, headerRef, contentRef })));
-      });
+      return onActionRegistered(actions => setActionConfigs(actions));
     }, [type]);
 
+    const discoveredActions = actionConfigs.map(action => (
```

Comment:
I needed to make this refactoring, because otherwise there was an unexpected memoization from React, which prevented update callback from firing